packages/mobile/src/localCurrency/sagas.ts

```diff
@@ -0,0 +1,74 @@
+import gql from 'graphql-tag'
+import { call, delay, put, spawn } from 'redux-saga/effects'
+import { apolloClient } from 'src/apollo'
+import { waitForRehydrate } from 'src/app/saga'
+import { LOCAL_CURRENCY_SYMBOL } from 'src/config'
+import {
+  fetchCurrentRate,
+  fetchCurrentRateFailure,
+  fetchCurrentRateSuccess,
+} from 'src/localCurrency/actions'
+import Logger from 'src/utils/Logger'
+
+const TAG = 'localCurrency/saga'
+
+const POLLING_DELAY = 3600 * 1000 // 1 hour
+
+// Delay before retrying when an error is raised
+const RETRY_DELAY = 10 * 1000 // 10 secs
+
+async function fetchExchangeRate(symbol: string): Promise<number> {
+  const response = await apolloClient.query({
+    query: gql`
+    {    
+      currencyConversion(currencyCode: "${symbol}") {
+        rate
+      }
+    }
+  `,
+    fetchPolicy: 'network-only',
+    errorPolicy: 'all',
+  })
+
+  const { data } = response
+  if (typeof data !== 'object') {
+    throw new Error(`Invalid response data ${data}`)
+  }
+
+  const { currencyConversion } = data
+  if (typeof currencyConversion !== 'object') {
+    throw new Error(`Invalid response data ${data}`)
+  }
+
+  const { rate } = currencyConversion
+  if (typeof rate !== 'number') {
+    throw new Error(`Invalid response data ${data}`)
+  }
+
+  return rate
+}
+
+function* updateLocalCurrencyRate() {
+  if (!LOCAL_CURRENCY_SYMBOL) {
+    return
+  }
+
+  yield call(waitForRehydrate)
+
+  while (true) {
+    try {
+      yield put(fetchCurrentRate())
+      const rate = yield call(fetchExchangeRate, LOCAL_CURRENCY_SYMBOL)
+      yield put(fetchCurrentRateSuccess(rate))
+      yield delay(POLLING_DELAY)
```

Comment:
I don't think it's recommended to have sleep/delays more than a few seconds. Can I suggest we just only fetch the rate once per app load. This seems fine because we only have daily rates anyway.

Reply:
Yes you're right. I remember this being a problem on RN on Android.
I'd suggest leaving as is for now until we come up with the appropriate change.

Reply:
Would it be difficult to simply remove the loop and just have this run once?

Reply:
No :D but the retry logic on errors needs to be altered.

Reply:
I'm personally okay to lose the retry logic. WDYT?

Reply:
I'd rather not remove it as it leaves the possibility for the rate to not be fetched at all if the user starts the app without an internet connection.